### PR TITLE
[LW] Various Recommendation Changes

### PR DESCRIPTION
**Pull request by darkruby501**
- changes the scenario in use for recommendations from `recombee-personal` to `recombee-lesswrong-custom`
- adds "sparkleIcon" as something that can be used in tab names, not yet in use because saving it for a bigger "next launch"
- adds sparkleIcon to post items that are the result of recommendations
- adds some improved tracking of scenario of displayed items, and recommId to better group them

LessWrong 2024-05-03 17-22-38

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1207228037228988) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/posts/PostsItemIcons.tsx [hunk 1]
```diff
@@ -88,14 +91,17 @@ export const CuratedIcon = ({hasColor, classes}: {
...
   post: PostsBase,
   hideCuratedIcon?: boolean,
   hidePersonalIcon?: boolean
+  showRecommendationIcon?: boolean,
   classes: ClassesType<typeof styles>,
 }) => {
   const { OmegaIcon, LWTooltip, CuratedIcon, ForumIcon } = Components;
 
+
+
```
Comment (b0b3rt) [inline packages/lesswrong/components/posts/PostsItemIcons.tsx:104]: delete

File: packages/lesswrong/components/posts/RecombeePostsList.tsx [hunk 2]
```diff
@@ -72,14 +72,17 @@ export const stickiedPostTerms: PostsViewTerms = {
   forum: true
 };
 
-export const RecombeePostsList = ({ algorithm, settings, limit = 15, classes }: {
+export const RecombeePostsList = ({ algorithm, settings, limit = 15, showRecommendationsIcon = false, classes }: {
```
Comment (b0b3rt) [inline packages/lesswrong/components/posts/RecombeePostsList.tsx:75]: naming of `showRecommendationsIcon` doesn't match `showRecommendationIcon` elsewhere (has an extra `s`)

File: packages/lesswrong/components/posts/RecombeePostsList.tsx [hunk 3]
```diff
@@ -98,13 +101,28 @@ export const RecombeePostsList = ({ algorithm, settings, limit = 15, classes }:
   });
 
   const results: RecommendedPost[] | undefined = data?.[resolverName]?.results;
+
+  const recommId = uniq(results?.map(({recommId}) => recommId) ?? []).join('-');
```
Comment (b0b3rt) [inline packages/lesswrong/components/posts/RecombeePostsList.tsx:105]: surely you want the recommIds to be inside of each item in `postIdsWithScenario`?  Or at least map the recommId to the scenario (so there's some way of associating them later)?

File: packages/lesswrong/components/posts/RecombeePostsList.tsx [hunk 4]
```diff
@@ -119,7 +122,7 @@ export const RecombeePostsList = ({ algorithm, settings, limit = 15, classes }:
   useOnMountTracking({
     eventType: "postList",
     // TODO: Remove postIds which is redundant once analytics dashboard written to use postIdsWithScenario
-    eventProps: { postIds, postIdsWithScenario, algorithm },
+    eventProps: { postIds, postIdsWithScenario, algorithm, recommId },
```
Comment (b0b3rt) [inline packages/lesswrong/components/posts/RecombeePostsList.tsx:125]: this probably also wants the vertex attributionId

File: packages/lesswrong/server/recombee/client.ts [hunk 5]
```diff
@@ -447,7 +447,7 @@ const recombeeApi = {
 
     let deferredPostsPromise: Promise<DbPost[]> | undefined = undefined;
     let recombeeResponsesWithScenario;
-    if (lwAlgoSettings.hybridScenarios.fixed === 'recombee-emulate-hacker-news') {
+    if (lwAlgoSettings.hybridScenarios.fixed === 'lesswrong-hacker-news') {
```
Comment (b0b3rt) [inline packages/lesswrong/server/recombee/client.ts:450]: If we're changing this, shouldn't it be to `forum-classic`?

Comment (darkruby501) [reply packages/lesswrong/server/recombee/client.ts:450]: Yeah, that makes sense

**Review by b0b3rt — APPROVED** [review]

**Review by jpaddison3 — COMMENTED** [review]
No objections

**Pull request merged**